frontend/src/v5/services/routing/routing.tsx

```diff
@@ -88,12 +88,11 @@ const WrapAuthenticationRedirect = ({ children }) => {
 
 	const location = useLocation();
 
-	AuthActionsDispatchers.setReturnUrl(location);
-
 	useEffect(() => {
 		if (!isAuthenticated && authenticationFetched) {
 			history.replace(LOGIN_PATH);
 		}
+		AuthActionsDispatchers.setReturnUrl(location);
```

Comment:
this should go before the if. is meant to save the return url, if you change the return url after you redirected it will save the wrong one.